File: Projetos/ACBrLib/Demos/C#/Shared/ACBrLib.Core/Ini/ACBrIniSection.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text;

namespace ACBrLib.Core
{
    public sealed class ACBrIniSection : Dictionary<string, string>
    {
        #region Constructors

        public ACBrIniSection(string name) : this(null, name)
        {
        }

        public ACBrIniSection(ACBrIniFile parent, string name)
        {
            Parent = parent;
            Name = name;
        }

        #endregion Constructors

        #region Properties

        public ACBrIniFile Parent { get; internal set; }

        public string Name { get; set; }

        #endregion Properties

        #region Methods

        internal void Save(TextWriter stream)
        {
            stream.WriteLine($"[{Name}]");

            foreach (var iniData in this)
                stream.WriteLine($"{iniData.Key}={iniData.Value}");

            stream.WriteLine("");
        }

        public TType GetValue<TType>(string key, TType defaultValue = default)
        {
            if (string.IsNullOrEmpty(key) || string.IsNullOrWhiteSpace(key)) return defaultValue;

            TType ret;
            try
            {
                if (!ContainsKey(key)) return defaultValue;

                ret = IniValueWrapper.UnWrap(this[key], defaultValue);
            }
            catch (Exception)
            {
                ret = defaultValue;
            }

            return ret;
        }

        public void SetValue<TType>(string key, TType value)
        {
            Add(key, IniValueWrapper.Wrap<TType>(value));
        }

        public override string ToString()
        {
            var builder = new StringBuilder();

            using (var writer = new StringWriter(builder))
                Save(writer);

            return builder.ToString();
        }

        #endregion Methods
    }
}

File: Projetos/ACBrLib/Android/Comum/ACBrLibComum/src/main/java/br/com/acbr/lib/comum/ini/ACBrIniSection.java
package br.com.acbr.lib.comum.ini;

import java.io.IOException;
import java.io.StringWriter;
import java.io.Writer;
import java.util.HashMap;

public class ACBrIniSection extends HashMap<String, String> {

    private ACBrIniFile parent;
    private String name;

    public ACBrIniSection(String name) {
        this(null, name);
    }

    public ACBrIniSection(ACBrIniFile parent, String name) {
        this.parent = parent;
        this.name = name;
    }

    public ACBrIniFile getParent() {
        return parent;
    }

    public void setParent(ACBrIniFile parent) {
        this.parent = parent;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    protected void save(Writer stream) throws IOException {
        stream.write("[" + name + "]\n");
        for (Entry<String, String> iniData : this.entrySet()) {
            stream.write(iniData.getKey() + "=" + iniData.getValue() + "\n");
        }
        stream.write("\n");
    }

    public <TType> TType getValue(String key, TType defaultValue) {
        if (key == null || key.trim().isEmpty()) {
            return defaultValue;
        }

        TType result;
        try {
            if (!containsKey(key)) {
                return defaultValue;
            }
            result = IniValueWrapper.unwrap(this.get(key), defaultValue);
        } catch (Exception e) {
            result = defaultValue;
        }
        return result;
    }

    public <TType> void setValue(String key, TType value) {
        put(key, IniValueWrapper.wrap(value));
    }

    @Override
    public String toString() {
        StringWriter writer = new StringWriter();
        try {
            save(writer);
        } catch (IOException e) {
            e.printStackTrace();
        }
        return writer.toString();
    }
}
